packages/dev/loaders/src/glTF/2.0/Extensions/objectModelMapping.ts

```diff
@@ -0,0 +1,1081 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import type { TransformNode } from "core/Meshes/transformNode";
+import type { IAnimation, ICamera, IGLTF, IKHRLightsPunctual_Light, IMaterial, IMesh, INode } from "../glTFLoaderInterfaces";
+import type { Vector3 } from "core/Maths/math.vector";
+import { Matrix, Quaternion, Vector2 } from "core/Maths/math.vector";
+import { Constants } from "core/Engines/constants";
+import type { Color3 } from "core/Maths/math.color";
+import { Color4 } from "core/Maths/math.color";
+import type { PBRMaterial } from "core/Materials/PBR/pbrMaterial";
+import type { Light } from "core/Lights/light";
+import type { Nullable } from "core/types";
+import { SpotLight } from "core/Lights/spotLight";
+import type { IEXTLightsImageBased_LightImageBased } from "babylonjs-gltf2interface";
+import type { BaseTexture } from "core/Materials/Textures/baseTexture";
+import type { IInterpolationPropertyInfo, IObjectAccessor } from "core/FlowGraph/typeDefinitions";
+import { GLTFPathToObjectConverter } from "./gltfPathToObjectConverter";
+
+export interface IGLTFObjectModelTree {
+    cameras: IGLTFObjectModelTreeCamerasObject;
+    nodes: IGLTFObjectModelTreeNodesObject;
+    materials: IGLTFObjectModelTreeMaterialsObject;
+    extensions: IGLTFObjectModelTreeExtensionsObject;
+    animations: {};
+    meshes: {};
+}
```

Comment:
I don't understand the types for this. What does this do?

Reply:
This is required for populating the object and having "length" implemented. I changed the types.